xlators/cluster/afr/src/afr-common.c

```diff
@@ -4161,11 +4153,6 @@ __afr_fd_ctx_set(xlator_t *this, fd_t *fd)
 
     priv = this->private;
 
-    ret = __fd_ctx_get(fd, this, &ctx);
-
-    if (ret == 0)
-        goto out;
-
     fd_ctx = GF_CALLOC(1, sizeof(afr_fd_ctx_t), gf_afr_mt_afr_fd_ctx_t);
     if (!fd_ctx) {
         ret = -ENOMEM;
```

Comment:
since you are not returning ret, no point in setting it. I think you can just return NULL here. Unsure if there's any cleanup that needs to happen in out, so you can goto out too.

And perhaps you can get rid of 'ret' entirely?